### DISCUSSION TITLE
Icon service HTTPS proxy?

### DISCUSSION THREAD

**Discussion by eiqnepm** [dani-garcia/vaultwarden#5049] [Q&A] [answered]
Is this possible? I know I could put Vaultwarden behind Gluetun, but this seems overkill, my VPN provides a HTTPS proxy which would be far more convenient to help against icon service IP leaks.

**Comment by BlackDex** [comment]
Have seen these options?

https://github.com/dani-garcia/vaultwarden/blob/040e2a7bb0f2cc5012d46ca99283cf21fa06ed1a/.env.template#L299...L330

You can also not use the built-in favicon fetching and use Google or even Bitwarden.

Else, Vaultwarden supports the HTTPS_PROXY/HTTP_PROXY env variables.

**Reply by eiqnepm** [reply]
> Have seen these options?
> 
> https://github.com/dani-garcia/vaultwarden/blob/040e2a7bb0f2cc5012d46ca99283cf21fa06ed1a/.env.template#L299...L330
> 
> You can also not use the built-in favicon fetching and use Google or even Bitwarden.
> 
> Else, Vaultwarden supports the HTTPS_PROXY/HTTP_PROXY env variables.
> 

I have, but I wanted to use the internal service behind a proxy.

I didn't see these environment variables in the template. What format does the `HTTPS_PROXY` variable require? Will `https://username:password@example.com:443` work?

**Reply by BlackDex** [reply, marked as answer]
The official docs can be found here: https://docs.rs/reqwest/latest/reqwest/#proxies and here https://docs.rs/reqwest/latest/reqwest/struct.Proxy.html

This looks useful too: https://gist.github.com/yougg/5d2b3353fc5e197a0917aae0b3287d64

**Reply by eiqnepm** [reply]
> The official docs can be found here: https://docs.rs/reqwest/latest/reqwest/#proxies and here https://docs.rs/reqwest/latest/reqwest/struct.Proxy.html
> 
> This looks useful too: https://gist.github.com/yougg/5d2b3353fc5e197a0917aae0b3287d64

Thanks!
